### PR TITLE
Refactor Offer DTO fields into required and optional parameters

### PR DESCRIPTION
**Pull request by Steffengreiner**
**Purpose of this PR**
This PR refactors the fields of the Offer DTO into required and optional parameters with default values for the offer building process as was requested here: #85 

**Misc**
I'm unsure which fields should be required and optional and why this is necessary in the first place. 
I'd rather argue, that we should avoid allowing offer creation without having all fields set, since it's a legal binding document without all the necesary information at hand.

### CODE REVIEW

**Comment by Steffengreiner** [comment]
I reduced the necessary amount of parameters for the Offer DTO to the most basic requirements:
`offerId` to ensure each created Offer is unique and can be tracked
`projectManager` to ensure accountability for a created Offer.  
`modificationDate` to track when the Offer was created
Let me know if you think this is too little or if other fields are more paramount to be set as required

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        Customer customer
+        ProjectManager projectManager
+        String projectTitle
+        String projectDescription
+        List<ProductItem> items
+        double totalPrice
+        CostEstimateId identifier
+        Affiliation selectedCustomerAffiliation
+
+        Builder(Date modificationDate) {
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy:72]: I think the modification date is not really relevant if you think about transferring offer information from the view to the use case. I would suggest having the project title, project description, customer, customer affiliation, PM, and items

File: src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy [hunk 2]
```diff
@@ -67,21 +68,53 @@ class Offer {
         OfferId identifier
         Affiliation selectedCustomerAffiliation
 
-        Builder(Date modificationDate, Date expirationDate, Customer customer, ProjectManager projectManager, String projectDescription, String projectTitle, List<ProductItem> items, double totalPrice, OfferId identifier, Affiliation selectedCustomerAffiliation) {
-            this.modificationDate = modificationDate
+        Builder(Date modificationDate) {
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy:71]: Same as for cost estimate

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
I would argue that we do need at least the minimal arguments to create an offer (as listed in one of my comments). The ID is something created by the use case and is added later to the information of the offer.

Other than that, thanks for taking care of the issue that I opened 👍

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy (hunk 1 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy:72]: Alright I can do that 👍

File: src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy [hunk 3]
```diff
@@ -28,7 +28,7 @@ class Offer {
     /**
      * The QBiC project manager who was assigned to the project
      */
-    final Person projectManager
+    final ProjectManager projectManager
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/dtos/business/Offer.groovy:31]: 🚀

**Review by jenniferboedker — APPROVED** [review]
LGTM

**Pull request merged**